resolver/src/main/java/io/netty/resolver/DefaultHostsFileEntriesResolver.java

```diff
@@ -24,42 +24,36 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Default {@link HostsFileEntriesResolver} that resolves hosts file entries only once.
  */
 public final class DefaultHostsFileEntriesResolver implements HostsFileEntriesResolver {
 
-    private final Map<String, List<InetAddress>> inet4Entries;
-    private final Map<String, List<InetAddress>> inet6Entries;
+    private final long refreshInterval;
+    private final AtomicLong lastRefresh = new AtomicLong(System.nanoTime());
+    private final HostsFileEntriesProvider.Parser hostsFileParser;
+    private volatile Map<String, List<InetAddress>> inet4Entries;
+    private volatile Map<String, List<InetAddress>> inet6Entries;
 
     public DefaultHostsFileEntriesResolver() {
-        this(parseEntries());
+        this(HostsFileEntriesProvider.parser(), TimeUnit.SECONDS.toNanos(10));
```

Comment:
Honestly I think every 10 seconds might be a bit aggressive... What about 60 seconds as default + maybe a system property to change it .

Reply:
I just wanted to add the question as a separate comment, didn't expect you will be so fast!

Reply:
I wasn't able to find a typical pattern across system property names, so pls let me know if I need to change it.